tests/foreman/cli/test_content_credentials.py

```diff
@@ -0,0 +1,880 @@
+Satellite 6.8
+
+:Requirement: ContentCredential
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ContentCredentials
+
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from tempfile import mkstemp
+
+from fauxfactory import gen_alphanumeric
+from fauxfactory import gen_choice
+from fauxfactory import gen_integer
+from fauxfactory import gen_string
+
+from robottelo import ssh
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.content_credentials import ContentCredential
+from robottelo.cli.factory import CLIFactoryError
+from robottelo.cli.factory import make_content_credential
+from robottelo.cli.factory import make_org
+from robottelo.cli.factory import make_product
+from robottelo.cli.factory import make_repository
+from robottelo.cli.org import Org
+from robottelo.cli.product import Product
+from robottelo.cli.repository import Repository
+from robottelo.constants import DEFAULT_ORG
+from robottelo.constants import VALID_GPG_KEY_FILE
+from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import valid_data_list
+from robottelo.decorators import tier1
+from robottelo.decorators import tier2
+from robottelo.decorators import upgrade
+from robottelo.helpers import get_data_file
+from robottelo.test import CLITestCase
+
+VALID_GPG_KEY_FILE_PATH = get_data_file(VALID_GPG_KEY_FILE)
+
+
+def create_gpg_key_file(content=None):
+    """Creates a fake GPG Key file and returns its path or None if an error
+    happens.
+    """
+
+    (_, key_filename) = mkstemp(text=True)
+    if not content:
+        content = gen_alphanumeric(gen_integer(20, 50))
+    with open(key_filename, "w") as gpg_key_file:
+        gpg_key_file.write(content)
+        return key_filename
+
+    return None
+
+
+class TestGPGKey(CLITestCase):
+    """Tests for GPG Keys via Hammer CLI"""
+
+    search_key = 'name'
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a shared organization for all tests to avoid generating
+        hundreds of organizations
+        """
+        super(TestGPGKey, cls).setUpClass()
+        cls.org = make_org(cached=True)
+
+    # Bug verification
+
+    @tier1
+    def test_verify_redmine_4272(self):
+        """content-credentials info should display key content
+
+        :id: 0ee87ee0-8bf1-4d15-b5f9-0ac364e61155
+
+        :expectedresults: content-credentials info should display key content
+
+        :CaseImportance: Critical
+        """
+        # Setup a new key file
+        content = gen_alphanumeric()
+        gpg_key = create_gpg_key_file(content=content)
+        self.assertIsNotNone(gpg_key, 'GPG Key file must be created')
+        gpg_key = make_content_credential(
+            {'key': gpg_key, 'name': gen_string('alpha'), 'organization-id': self.org['id']}
+        )
+        self.assertEqual(gpg_key['content'], content)
+
+    @tier1
+    def test_positive_get_info_by_name(self):
```

Comment:
I think the cases here could be simplified if you wanted to reduce the duplication of actions.

These initial cases are doing slight variations of the same thing, and appear to have a fair amount of duplication in what's being created and checked for existence. Like this get info test could be done at the same time as any of the `exists` checks that are using various name, key, and organization values though unittest subtest.

Reply:
@mshriver as I explained above, these are just clones of existing gpg key tests to try and get it done ASAP as the new sub-command was only discovered by myself recently.